### PR TITLE
Add napi.update_event

### PR DESCRIPTION
**Pull request by awarecan**
So that the client can listen the update from Nest Stream API

~~Also included change for jkoelker/python-nest #132~~

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,8 @@ develop-eggs
 .installed.cfg
 lib
 lib64
+share
```
Comment (theaquarium) [inline .gitignore]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (theaquarium) [inline .vscode/settings.json]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: README.rst [hunk 3]
```diff
@@ -230,8 +230,8 @@ A configuration file must be specified and used for the credentials to communica
 .. code-block:: ini
 
     [NEST]
-    user = joe@user.com
-    password = swordfish
+    client_id = yourclientid
```
Comment (theaquarium) [inline README.rst]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: nest/nest.py [hunk 4]
```diff
@@ -1331,6 +1331,10 @@ def snapshot_url(self):
         else:
             return SIMULATOR_SNAPSHOT_PLACEHOLDER_URL
 
+    @property
```
Comment (theaquarium) [inline nest/nest.py]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: nest/command_line.py [hunk 5]
```diff
@@ -148,12 +148,14 @@ def get_device(napi, args, structure):
         return structure.cameras[args.index]
 
 
-def handle_camera_show(device):
-    print('Device                : %s' % device.name)
-    # print('Model               : %s' % device.model) # Doesn't seem to work
-    print('Serial                : %s' % device.serial)
-    print('Where                 : %s' % device.where)
-    print('Where ID              : %s' % device.where_id)
+def handle_camera_show(device, count=0):
```
Comment (theaquarium) [inline nest/command_line.py]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: nest/command_line.py [hunk 6]
```diff
@@ -227,6 +279,11 @@ def _identity(x):
             pin = input("PIN: ")
             napi.request_token(pin)
 
+        if not cmd:
```
Comment (theaquarium) [inline nest/command_line.py]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: nest/command_line.py [hunk 7]
```diff
@@ -334,43 +391,11 @@ def _identity(x):
...
-                      (display_temp(device.target), device.temperature_scale))
-
-            helpers.print_if('Away Heat             : %0.1fC',
-                             device.eco_temperature[0])
-
-            helpers.print_if('Away Cool             : %0.1fC',
-                             device.eco_temperature[1])
-
-            print('Has Leaf              : %s' % device.has_leaf)
+            count = 0
```
Comment (theaquarium) [inline nest/command_line.py]: If this is not directly related to the streaming API, please submit changes into a pull request in the main repo.

File: nest/command_line.py (hunk 5 above)
Comment (awarecan) [reply nest/command_line.py]: I just want to include a showcase the usage of update_event and the stream API. I added a new argument to keep the command line tool behavior backward compatible. Does it make sense now?

**Comment by awarecan** [comment]
@theaquarium the change of command line is for demonstration the benefit of stream API. I added a new command line argument to keep the default behavior same as previous

File: nest/command_line.py (hunk 5 above)
Comment (theaquarium) [reply nest/command_line.py]: Thanks. This should work.

**Comment by awarecan** [comment]
In the camera-show and show command, it will refresh no matter which device's state updated. We can resolve that by add granular update_event for each device/sensor. However I feel that it will be out of scope  of this PR. I am willing to work on that part as long as this PR got merged.

**Comment by awarecan** [comment]
Change to use requests in sseclient, so that we got unified Authorization workflow, also add connection closed handling, will bake it tonight

**Comment by awarecan** [comment]
It worked well for overnight testing. After 14 hours, the connection been reset by server, and it successfully created a new connection

**Comment by theaquarium** [comment]
Awesome!
